pwiz_tools/Skyline/Util/Adduct.cs

```diff
@@ -702,6 +702,38 @@ public static Adduct ProtonatedFromFormulaDiff(string left, string right, int ch
             return new Adduct(string.Format(@"[M{0}{1}{2}H]", sign, adductFormula, charge>0?@"+":@"-"), ADDUCT_TYPE.non_proteomic) { AdductCharge = charge };
         }
 
+        /// <summary>
+        /// Changes text like "[M+H]" to just "M+H"
+        /// </summary>
+        public static string StripBrackets(string adductText)

```

Comment:
These functions just still look very loose to me, which is why it is good to pull them out like this. They don't look like they belong in an Adduct class, because all they do is tell if a formula contains brackets and strip all brackets out of any kind of text. I would expect them to expect more out of well-formed adduct text. Why not HasAdduct(formula) instead, and actually look for some kind of proper placement? Couldn't you just make everything based on SplitFormulaAndTrailingAdduct? Do that up front and then test string.IsNullOrEmpty(formulaAdduct), and then if you still feel you need to strip brackets to make an equals comparison... Write an Adduct.AreEqual() function instead?